fixed to compile on RHEL 7 based kernels

pmcgleenon opened the pull request:
the 3.13 kernel version checks aren't sufficient for the NF callback API for RHEL7 based kernels (hooknum is now a const struct nf_hook_ops) rather than int

RHEL7 and later advertises a kernel version 3.10, which has the later version of the NF callback.   the vanilla kernel introduced this later than 3.13.

I'm assuming the RHEL clones like CentOS, Oracle, Scientific Linux etc will have the same issue

tomerb committed:
Add support for kernel > v4.0.

In change 1d1de89b, nf_queue_entry was refactored, and since v4.0-rc6
some of its member variables have now been extracted to nf_hook_state.

tomerb committed:
In Linux 4.1 nf_hookfn()'s signature has changed.

pmcgleenon committed:
fixed to compile on RHEL 7 based kernels

pmcgleenon commented:
RHEL7_NF_BACKPORT is defined in kipfw/Makefile

compiles fine for me on RHEL 7.1.   a further patch is needed for RHEL 7.2 - the NF hook API has changed there too

FS1360472174 commented:
@pmcgleenon 
you have added the condition RHEL7.but still got error about the  define of  call_ipfw.
which call at ipfw2_mod.c line 848.
I test on  linux-3.10.0-327.4.5.el7.so how to fix it?

pmcgleenon committed:
Merge pull request #1 from tomerb/master

adding support for linux 4.x kernels

pmcgleenon commented:
So it looks like you've got the RHEL 7.2 kernel version.   RHEL 7.2 introduced a further change in the NF Hook API.   We'll need to make a change similar to https://github.com/luigirizzo/dummynet/pull/6  for RH7.2

FS1360472174 commented:
@pmcgleenon I have fixed the issue for RH7.2.
actually I dislike adding so many linux version condition on Hook type define.
we need to note that.as I mention at https://github.com/luigirizzo/dummynet/pull/13

pmcgleenon committed:
refactored to handle different netfilter API versions

Commit:
updated default max queue size

Commit:
Fixed kernel crash on RHEL 7.5 related to copy from user to kernel

pmcgleenon committed:
removed file

Commit:
delay profile: don't drop packet when drop_rate = 0

pmcgleenon committed:
added IP6 support

pmcgleenon committed:
fixed plr formatting

pmcgleenon committed:
updated README

pmcgleenon committed:
renamed README

pmcgleenon committed:
added README.md

pmcgleenon committed:
updated README

pmcgleenon committed:
updated README

pmcgleenon committed:
updated README

pmcgleenon committed:
updated README

pmcgleenon committed:
updated README

pmcgleenon committed:
Extended max queue size to 10MB

pmcgleenon committed:
increased slot limit to 9K

pmcgleenon committed:
Raised byte_limit and slot_limit to 50MB equivalent

pmcgleenon committed:
Increased max bw from max int (2147483647) to max uint (4294967295)

Similar change was applied to FreeBSD
https://github.com/pfsense/FreeBSD-src/commit/2085b4c32205d4b41c4cdc810db1b9531881c824#diff-8357b1b1701d22d67098d5b7219e9c6f

pmcgleenon committed:
Added jitter capability

pmcgleenon committed:
Create LICENSE

Adding license file
